src/twisted/names/test/test_client.py

```diff
@@ -528,6 +528,24 @@ def query(self, address, query, timeout=10, id=None):
         self.assertEqual(len(set(protocols)), 2)
 
 
+    def test_ipv6Resolver(self):
+        """
+        If the resolver is ipv6, open a ipv6 port.
+        """
+        counter = []
+
+        class FakeReactor(MemoryReactor):
```

Comment:
This particular tests has to subclass this particular `MemoryReactor` and call into its `listenUDP`, but `test_disallowedPort` does not.  That's because `test_disallowedPort` needs to make [this condition](https://github.com/twisted/twisted/blob/63f7b78854198fbc25e779b6e852b7a5dba2a6fd/src/twisted/names/dns.py#L2911) true to hit the [`except` block](https://github.com/twisted/twisted/blob/63f7b78854198fbc25e779b6e852b7a5dba2a6fd/src/twisted/names/dns.py#L2915-L2916).

_This_ test, however, exercises the _client_'s [`listenUDP`](https://github.com/twisted/twisted/pull/649/files#diff-7a7cb05a7b26a752daf0078ea416302fR215), so it needs to ensure the transport has been set on the `DNSDatagramProtocol`.  Doing that fails the condition linked to above, so that `startListening` is not called. `test_util.MemoryReactor.listenUDP` does that  by calling [`makeConnection` on its protocol](https://github.com/twisted/twisted/blob/63f7b78854198fbc25e779b6e852b7a5dba2a6fd/src/twisted/names/test/test_util.py#L134)

I have three opinions after writing this story:

1. I'd suggest importing `test_util` so that the superclass name would read `test_util.MemoryReactor`.  We have two `MemoryReactor`s running around and since it's critical that the one inside `twisted.names` be used here, it'd be nice to make its source explicit.
2. You might consider returning the result of `listenUDP` just to stay consistent with `test_util.MemoryReactor.listenUDP`'s interface.
3. Maybe a comment or docstring explaining why this particular `MemoryReactor` is critical to the test?